application/src/main/scala/com/azavea/franklin/database/Filterables.scala

```diff
@@ -25,6 +28,34 @@ trait FilterHelpers {
       }
     }
   }
+
+  implicit class QueryWithFilter(query: Query) {
+
+    def toFilterFragment(field: String) = {
+      val fieldFragment = Fragment.const(s"item -> 'properties' -> '$field'")
+      query match {
+        case Equals(value) =>
+          fieldFragment ++ fr"= $value"
+        case NotEqualTo(value) =>
+          fieldFragment ++ fr"<> $value"
+        case GreaterThan(floor) =>
+          fieldFragment ++ fr"> $floor"
+        case GreaterThanEqual(floor) =>
+          fieldFragment ++ fr">= $floor"
+        case LessThan(ceiling) =>
+          fieldFragment ++ fr"< $ceiling"
+        case LessThanEqual(ceiling) =>
+          fieldFragment ++ fr"<= $ceiling"
+        case StartsWith(prefix) =>
+          Fragment.const(s"starts_with(item ->> '$field', '$prefix')")
+        case EndsWith(postfix) =>
+          Fragment.const(s"right(item ->> '$field', ${postfix.value.length})") ++ fr"= $postfix"
+        case Contains(substring) =>
+          Fragment.const(s"strpos(item ->> '$field', '$substring') > 0")
+        case In(values) => Fragments.in(fieldFragment, values)
```

Comment:
* works: `{"query":{"layer:ids":{"eq": ["layer-us", "layer-ca"]}}}` 
* fails: `{"query":{"layer:ids":{"in": ["layer-us", "layer-ca"]}}}`

P.S. what are the semantics of the `in`? I guess it is an expected behavior; but still wondering.

Reply:
Oh, hmm. I think there isn't actually an operator for the `layer:ids` query that you want. `in` at least appears to be for `elem a as`:

```
    "stringAttr3": {
      "in": ["landsat", "modis", "naip"]
    }
```

I'm reading `stringAttr3` that way at least. Maybe what we need is an `intersects` operator or something, for `arrayQuery \subset arrayAttr`.

Reply:
@jisantuc intersects sounds good to me! could you add it in terms of this PR please? 

Just for the testing purposes, that is the query that worked perfectly for me:

```sql
(item #> '{properties, "layer:ids"}') @> '["layer-us", "layer-ca"]'
```

Reply:
Yeah I'll add it and try to PR it into the actual extension. That went well last time... :cry: (in the sense that I still owe PR follow-up because a billion things got tacked onto my PR)

Reply:
A cool thing is that this will also make it possible to filter for specific band combinations with EO

Reply:
Yeah, I think intersects makes sense here too.

I looked up to see what predicates are in ["OGC API - Features - Part 3: Common Query Language"](http://docs.opengeospatial.org/DRAFTS/19-079.html#rc_cql-json) to see how they suggest to handle this query and I didn't see one. There _is_ an intersects operator for a spatial query, but not for other things.

I don't think that's a problem and we should implement it if we're going to use it. Do you think we should open an issue https://github.com/opengeospatial/ogcapi-features/issues to see what they say?

Reply:
Oof, can't believe I didn't realize the collision with spatial filtering. Yeah I'll open an issue that asks how they'd suggest handling it and link this discussion. For now I'll name the operator `superset` I guess

Reply:
I confirmed that 155af82 makes this kind of filtering possible by messing with the array passed to `layers` in this json:

```json
{
    "bbox": null,
    "datetime": null,
    "intersects": null,
    "collections": null,
    "query": {
	"layers": {
	    "superset": ["layer-mex"]
	}
    }
}
```

and posting to `:9090/search`

Reply:
https://github.com/opengeospatial/ogcapi-features/issues/384